Yapf ImportError

### Summary

Yapf is failing on github actions.

Looks like this is fixed upstream in 0.40.3

### Steps To Reproduce

<!--- Please provide detailed steps for reproducing this issue. -->

### Current Behaviour

<!--- Please provide information about the undesired current behaviour. -->

### Expected Behaviour

<!--- Explain how the program should behave once the issue has been resolved. -->

### Context

<!--- Please provide any relevant information about your system setup. Mantid Imaging version can be obtained by running `conda list mantidimaging` and system information can be obtained by running `uname -a`. -->

### Failure Logs

```
   yapf --parallel --diff --recursive .
  shell: /usr/bin/bash -l {0}
  env:
    INPUT_RUN_POST: true
    CONDA: /home/runner/miniconda3
##[debug]/usr/bin/bash -l /home/runner/work/_temp/27c6f8a0-8217-4a11-8cd5-0f71bca8314a.sh
Traceback (most recent call last):
  File "/home/runner/miniconda3/envs/mantidimaging-dev/bin/yapf", line 6, in <module>
    from yapf import run_main
  File "/home/runner/miniconda3/envs/mantidimaging-dev/lib/python3.12/site-packages/yapf/__init__.py", line 36, in <module>
    from importlib_metadata import metadata
ImportError: cannot import name 'metadata' from 'importlib_metadata' (unknown location)
Error: Process completed with exit code 1.
```


### Screenshot(s)

<!--- [Optional] Include screenshots that illustrate the problem. -->

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -5,7 +5,7 @@ repos:
     -   id: end-of-file-fixer
         types: [python]
 -   repo: https://github.com/google/yapf
-    rev: v0.40.2
+    rev: v0.43.0
     hooks:
     -   id: yapf
 -   repo: local

--- a/environment-dev.yml
+++ b/environment-dev.yml
@@ -11,7 +11,7 @@ dependencies:
   - pip
   - pip:
       - eyes-images==5.23.*
-  - yapf==0.40.*
+  - yapf==0.43.*
   - mypy==1.8
   - types-requests
   - types-PyYAML

